### ISSUE TITLE
Accessing _cachedSystemAnimationFence requires the main thread

### ISSUE THREAD

**Issue by raminious** [wkh237/react-native-fetch-blob#168]
I'm not sure this is react-native-fetch-blob error or not, please feel free to close this issue if error not depend on your module.

I use [react-native-router-flux](https://github.com/aksonov/react-native-router-flux/) as navigator of my app
When I use [Swtich](https://github.com/aksonov/react-native-router-flux/blob/master/docs/OTHER_INFO.md#switch-new-feature) (for example login and swtich to application) and want to open file with `openDocument` I will receive this error:

![image](https://cloud.githubusercontent.com/assets/334716/19836623/65c21d3c-9ebb-11e6-95e8-997f50290a10.png)

EDIT 1:
I changed the logic of app and removed Swtich, but still get above error when changing Scene

**Comment by raminious** [comment]
I finally found the problem
This error is happening when I use ```autoFocus``` in ```<TextInput />``` component !

**Comment by raminious** [comment]
@wkh237
I still experience this error in many scenarios
Do you have any idea ?

**Comment by wkh237** [comment]
@raminious , thanks for the information, I'm not be able to reproduce the bug. Could you prvode a code snippet ? that'd be helpful 👍

**Comment by raminious** [comment]
Thank you @wkh237 

Please run this snippet code I created https://gist.github.com/raminious/60f9183f1875023ec7c819203681e89c

Open app in your iOS simulator and paste a video link to download (sample video: http://techslides.com/demos/sample-videos/small.mp4)
then click on button to open the video. it should throw an error about main thread

error description is a little different with previous error I mentioned but I think these two errors are same.

![image](https://cloud.githubusercontent.com/assets/334716/19945117/edf596bc-a153-11e6-8ce6-cc62eb30f90e.png)

**Comment by wkh237** [comment]
@raminious , thanks for the great information, I'll look into it.

**Comment by wkh237** [comment]
I've tried your code snippet but unfortunately I couldn't reproduce the error, what's the RN and RNFB's version ?

<img src="https://cloud.githubusercontent.com/assets/5063785/19954331/504593cc-a1b2-11e6-85dc-aa8dd2093d0c.gif" width="320"/>

**Comment by raminious** [comment]
Thank you @wkh237 


https://gist.github.com/raminious/214a3cf9edb19523eda047c857004128

![image](https://cloud.githubusercontent.com/assets/334716/19960048/b3d79144-a1bf-11e6-9ec5-178ca3bc5bdf.png)

![kapture 2016-11-03 at 12 15 22](https://cloud.githubusercontent.com/assets/334716/19960010/890984c2-a1bf-11e6-8a2a-8659d64b8c17.gif)

**Comment by raminious** [comment]
Hi @wkh237 
Is this issue fixable ?

**Comment by wkh237** [comment]
@raminious , I've created a clean RN 0.36 environment but still can't make it happen. Could you try adding these line and test it in you environment ?

RNFetchBlob.m line 458

```diff
RCT_EXPORT_METHOD(openDocument:(NSString*)uri scheme:(NSString *)scheme resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject
{
    
    NSURL * url = [[NSURL alloc] initWithString:uri];
    documentController = [UIDocumentInteractionController interactionControllerWithURL:url];
    documentController.delegate = self;
    
    if(scheme == nil || [[UIApplication sharedApplication] canOpenURL:[NSURL URLWithString:scheme]]) {
+        dispatch_sync(dispatch_get_main_queue(), ^{                                                                                                                                                                                                                                    
             [documentController presentPreviewAnimated:YES];
+        });                                                                                                                                                                                                                                    
        resolve(@[[NSNull null]]);
    } else {
        reject(@"RNFetchBlob could not open document", @"scheme is not supported", nil);
    }
})

```

RNFetchBlob.m line 441

```diff
RCT_EXPORT_METHOD(previewDocument:(NSString*)uri scheme:(NSString *)scheme resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject
{
    
    NSURL * url = [[NSURL alloc] initWithString:uri];
    documentController = [UIDocumentInteractionController interactionControllerWithURL:url];
    UIViewController *rootCtrl = [[[[UIApplication sharedApplication] delegate] window] rootViewController];
    documentController.delegate = self;
    if(scheme == nil || [[UIApplication sharedApplication] canOpenURL:[NSURL URLWithString:scheme]]) {
+       dispatch_sync(dispatch_get_main_queue(), ^{                                                                                                                                                                                                                                    
            [documentController  presentOptionsMenuFromRect:rootCtrl.view.bounds inView:rootCtrl.view animated:YES];
+       });                                                                                                                                                                                                                                    
        resolve(@[[NSNull null]]);
    } else {
        reject(@"RNFetchBlob could not open document", @"scheme is not supported", nil);
    }
})
```

**Comment by raminious** [comment]
@wkh237 omg thats it
It finally solved bug 👍

**Comment by wkh237** [comment]
Great to know that, I'll add these lines to IOS module and publish another release.

**Comment by raminious** [comment]
Thank you @wkh237 
Im still thinking why you don't face this error. may be this error throws on iOS 10

**Comment by wkh237** [comment]
You're welcome, the fix has launched in `0.10.0`, please upgrade if needed 👍

**Comment by raminious** [comment]
Awesome
Thanks 👍

**Comment by klinkov** [comment]
Did not find this your fix in last react-native-fetch-blob@0.10.0-beta.4
Please, fix it!
